### PR TITLE
enhancement(integration): Mixpanel event mapping update for track

### PR DESCRIPTION
**Pull request by shrouti1507**
## Description of the change

- enhanced the mapping for `track` events based on user request. i.e, `context.app.build` to `$app_build_number` and - 
   `context.app.version`to `$app_version_string` and `messageId` to `$insert_id`
-  checked for the existence of `context` fields for every if statement, where we are trying to access any field inside `message.context`
- edited test case accordingly
- added doc ref to the implementation, which was missing
- added error handling for incorrect `$geo_source` value

Screenshot 2022-07-04 at 11 51 33 AM

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in develop

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by utsabc** [comment]
Whats the situation with device mode?

**Comment by utsabc** [comment]
Also we have to update the test cases were the test cases earlier not capturing these fields?

**Comment by shrouti1507** [comment]
> Also we have to update the test cases were the test cases earlier not capturing these fields?

We have added test cases, hence the results are changed.

**Comment by shrouti1507** [comment]
> Whats the situation with device mode?

In device mode, we are not explicitly mapping anything for build and app_version. Also in their documentation [here](https://developer.mixpanel.com/docs/javascript#track-events) and [here](https://developer.mixpanel.com/docs/javascript-full-api-reference#mixpaneltrack), not finding any scope to add that.

>

**Review by arnab-p — CHANGES_REQUESTED** [review]
Let's review all the mappings before merging this

File: v0/destinations/mp/data/MPEventPropertiesConfig.json [hunk 1]
```diff
@@ -112,15 +115,24 @@
...
+    "sourceKeys": [
+      "properties.appVersionString",
+      "context.app.version"
+    ],
     "destKey": "$app_version_string"
   },
   {
-    "sourceKeys": "properties.brand",
+    "sourceKeys": [
+      "device.manufacturer",
```
Comment (utsabc) [inline v0/destinations/mp/data/MPEventPropertiesConfig.json:133]: Lets keep the original ordering

File: v0/destinations/mp/data/MPEventPropertiesConfig.json [hunk 2]
```diff
@@ -150,5 +162,13 @@
...
     "sourceKeys": "properties.googlePlayServices",
     "destKey": "$google_play_services"
+  },
+  {
+    "sourceKeys": "messageId",
+    "destKey": "$insert_id"
+  },
+  {
+    "sourceKeys": "properties.mpApiEndpoint",
+    "destKey": "$mp_api_endpoint"
```
Comment (utsabc) [inline v0/destinations/mp/data/MPEventPropertiesConfig.json:172]: This value should be set to host header ref:
Mp Api Endpoint (raw property name: $mp_api_endpoint)
This is the value of the [Host Header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host) based on the request to our ingestion service. This value will commonly be api.mixpanel.com. However, it is useful to confirm where your data was ingested if you partake in our [EU Data Residency](https://help.mixpanel.com/hc/en-us/articles/360039135652-Data-Residency-in-EU) or are using a custom proxy.

We don't need to set this from property, lets omit this field

File: v0/destinations/mp/data/MPProfilePropertiesAndroid.json [hunk 3]
```diff
@@ -28,7 +28,10 @@
     "destKey": "$android_lib_version"
   },
   {
-    "sourceKeys": "properties.androidBrand",
+    "sourceKeys": [
+      "context.device.manufacturer",
```
Comment (utsabc) [inline v0/destinations/mp/data/MPProfilePropertiesAndroid.json:32]: Lets keep original order for backward compatibility

**Comment by utsabc** [comment]
For test case can we do base64 decode at test.js file to have better visibility when matching? Also can we update the test cases to ga4 format?

**Comment by shrouti1507** [comment]
> Let's review all the mappings before merging this

checked and addressed

**Comment by shrouti1507** [comment]
> For test case can we do base64 decode at test.js file to have better visibility when matching? Also can we update the test cases to ga4 format?

addressed the visibility part. and as discussed, GA4 like test cases will be created later on

**Review by utsabc — APPROVED** [review]

File: v0/destinations/mp/data/MPProfilePropertiesAndroid.json (hunk 3 above)
Comment (shrouti1507) [reply v0/destinations/mp/data/MPProfilePropertiesAndroid.json]: done

File: v0/destinations/mp/data/MPEventPropertiesConfig.json (hunk 2 above)
Comment (shrouti1507) [reply v0/destinations/mp/data/MPEventPropertiesConfig.json]: removed

File: v0/destinations/mp/data/MPEventPropertiesConfig.json (hunk 1 above)
Comment (shrouti1507) [reply v0/destinations/mp/data/MPEventPropertiesConfig.json]: changed

File: v0/destinations/mp/data/MPEventPropertiesConfig.json [hunk 4]
```diff
@@ -112,15 +115,24 @@
...
   },
   {
-    "sourceKeys": "properties.brand",
+    "sourceKeys": [
+      "properties.brand",
+      "device.manufacturer"
+    ],
     "destKey": "$brand"
   },
   {
```
Comment (utsabc) [inline v0/destinations/mp/data/MPEventPropertiesConfig.json:138]: for `$radio` its written
Radio ($radio)
Current [cellular network communication standard](https://www.objc.io/issues/5-ios7/iOS7-hidden-gems-and-workarounds/#know-your-radio) (3G, 4G, LTE, etc).

We should be able to use carrier info for this

**Review by arnab-p — APPROVED** [review]

**Pull request merged**